lib/srv/db/redis/engine.go

```diff
@@ -0,0 +1,263 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package redis
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"net"
+
+	"github.com/go-redis/redis/v8"
+	"github.com/gravitational/teleport/lib/defaults"
+	"github.com/gravitational/teleport/lib/services"
+	"github.com/gravitational/teleport/lib/srv/db/common"
+	"github.com/gravitational/teleport/lib/srv/db/common/role"
+	"github.com/gravitational/teleport/lib/srv/db/redis/protocol"
+	"github.com/gravitational/teleport/lib/utils"
+	"github.com/gravitational/trace"
+)
+
+func init() {
+	common.RegisterEngine(newEngine, defaults.ProtocolRedis)
+}
+
+// newEngine create new Redis engine.
+func newEngine(ec common.EngineConfig) common.Engine {
+	return &Engine{
+		EngineConfig: ec,
+	}
+}
+
+// Engine implements common.Engine.
+type Engine struct {
+	// EngineConfig is the common database engine configuration.
+	common.EngineConfig
+	// clientConn is a client connection.
+	clientConn net.Conn
+	// clientReader is a go-redis wrapper for Redis client connection.
+	clientReader *redis.Reader
+	// sessionCtx is current session context.
+	sessionCtx *common.Session
+}
+
+// InitializeConnection initializes the database connection.
+func (e *Engine) InitializeConnection(clientConn net.Conn, sessionCtx *common.Session) error {
+	e.clientConn = clientConn
+	e.clientReader = redis.NewReader(clientConn)
+	e.sessionCtx = sessionCtx
+
+	// Let client to not set the database user, as Redis uses default user "default" if a user
+	// is not provided.
```

Comment:
nit: Can probably just remove this comment as the code is clear.
```suggestion
	// Use Redis default user named "default" if a user is not provided.
```